main.go

```diff
@@ -204,55 +225,60 @@ func setupRayClusterController(mgr ctrl.Manager, cfg *config.CodeFlareOperatorCo
 	return rayClusterController.SetupWithManager(mgr)
 }
 
-// +kubebuilder:rbac:groups="apiextensions.k8s.io",resources=customresourcedefinitions,verbs=get;list;watch
-
 func waitForRayClusterAPIandSetupController(ctx context.Context, mgr ctrl.Manager, cfg *config.CodeFlareOperatorConfiguration, isOpenShift bool, certsReady chan struct{}) {
-	crdClient, err := apiextensionsclientset.NewForConfig(mgr.GetConfig())
-	exitOnError(err, "unable to create CRD client")
-
-	crdList, err := crdClient.ApiextensionsV1().CustomResourceDefinitions().List(ctx, metav1.ListOptions{})
-	exitOnError(err, "unable to list CRDs")
-
-	if slices.ContainsFunc(crdList.Items, func(crd apiextensionsv1.CustomResourceDefinition) bool {
-		return crd.Name == "rayclusters.ray.io"
-	}) {
+	if isAPIAvailable(ctx, mgr, rayclusterAPI) {
 		exitOnError(setupRayClusterController(mgr, cfg, isOpenShift, certsReady), "unable to setup RayCluster controller")
+	} else {
+		waitForAPI(ctx, mgr, rayclusterAPI, func() {
+			exitOnError(setupRayClusterController(mgr, cfg, isOpenShift, certsReady), "unable to setup RayCluster controller")
+		})
 	}
+}
 
-	retryWatcher, err := retrywatch.NewRetryWatcher(crdList.ResourceVersion, &cache.ListWatch{
-		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
-			return crdClient.ApiextensionsV1().CustomResourceDefinitions().List(ctx, metav1.ListOptions{})
-		},
-		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
-			return crdClient.ApiextensionsV1().CustomResourceDefinitions().Watch(ctx, metav1.ListOptions{})
-		},
-	})
-	exitOnError(err, "unable to create retry watcher")
-
-	defer retryWatcher.Stop()
-	for {
-		select {
-		case <-ctx.Done():
-			return
-		case event := <-retryWatcher.ResultChan():
-			switch event.Type {
-			case watch.Error:
-				exitOnError(apierrors.FromObject(event.Object), "error watching for RayCluster API")
+func setupAppWrapperComponents(ctx context.Context, cancel context.CancelFunc, mgr ctrl.Manager,
+	cfg *config.CodeFlareOperatorConfiguration, certsReady chan struct{}) error {
+	if cfg.AppWrapper == nil || !ptr.Deref(cfg.AppWrapper.Enabled, false) {
+		setupLog.Info("Embedded AppWrapper controller is disabled by config")
+		externalController := cfg.AppWrapper != nil && ptr.Deref(cfg.AppWrapper.ExternalController, false)
```

Comment:
I'm not sure I get the context behind that `externalController` configuration parameter. What's the added value compared to setting the mock webhook when the appwrapper integration is not enabled?

Reply:
simplified per discussion this morning.   Installing a mock webhook will be sufficient to enable an external controller.